### PR TITLE
revisions to issue templates (and formalization of process for soliciting bugs and feature requests)

### PR DESCRIPTION
**Pull request by neuromusic**
This PR adds issue templates to guide users where and how to submit feedback on the napari hub.

One goal is to try to reserve "Issues" for engineering-ready work (or at least for committed work even if there is iteration and alignment needed on the details of the work).

## Questions
This PR adds a prompt when to submit questions to the discussion forum.

## Bugs
Bugs should be submitted as Github issues. ~This adds a template for bug reports.~

## Features
Feature requests should be submitted to our "Discussions" forum. From there, we can discuss with other users and stakeholders, dig in to understand the underlying problems and motivations, our team can explore solutions which may be different from the requested feature, and once we are confident that "there's a there there", we can build out user stories and designs to drive implementation.

This PR adds a prompt when to submit feature requests to the discussion forum.

## User Stories
Once we have identified new features that we want to build, our team will create User Stories to guide design and engineering work. 

This PR adds a template for user stories.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -1,2 +1,11 @@
 # Ref: https://help.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository#configuring-the-template-chooser
 blank_issues_enabled: true  # default
+contact_links:
+- name: 💡 Feature Request
+  url: https://github.com/chanzuckerberg/napari-hub/discussions/new
```
Comment (justinelarsen) [inline .github/ISSUE_TEMPLATE/config.yml:5]:
```suggestion
  url: https://github.com/chanzuckerberg/napari-hub/discussions/new?category=ideas
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -1,2 +1,11 @@
...
+  about: |
+  Have an feature request? Start an "Ideas" thread on our discussion forum.
+- name: 🤷 How do I?
+  url: https://github.com/chanzuckerberg/napari-hub/discussions/new
```
Comment (justinelarsen) [inline .github/ISSUE_TEMPLATE/config.yml:9]:
```suggestion
  url: https://github.com/chanzuckerberg/napari-hub/discussions/new?category=q-a
```

File: .github/ISSUE_TEMPLATE/user-story.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+### Acceptance Criteria
+
+1. [If I do A, B should happen.]
+
+[
+Also, here are a few points that need to be addressed:
+
+1. Constraint 1;
+1. Constraint 2;
```
Comment (justinelarsen) [inline .github/ISSUE_TEMPLATE/user-story.md:27]:
```suggestion
2. Constraint 2;
```

File: .github/ISSUE_TEMPLATE/user-story.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+1. Constraint 3.
```
Comment (justinelarsen) [inline .github/ISSUE_TEMPLATE/user-story.md:28]:
```suggestion
3. Constraint 3.
```

**Review by justinelarsen — COMMENTED** [review]
Looks good overall! I made a couple of suggestions. The PR description said this PR adds a template for bug reports, but I didn't see it.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 1 above)
Comment (neuromusic) [reply .github/ISSUE_TEMPLATE/config.yml:5]: yes! thank you for figuring this out!!

File: .github/ISSUE_TEMPLATE/user-story.md (hunk 3 above)
Comment (neuromusic) [reply .github/ISSUE_TEMPLATE/user-story.md:27]: markdown doesn't care, but you're right that this will help with legibility for the author

**Comment by neuromusic** [comment]
> The PR description said this PR adds a template for bug reports, but I didn't see it.

Woops, you're right, @justinelarsen ... the repo already had a template

File: .github/ISSUE_TEMPLATE/config.yml [hunk 5]
```diff
@@ -1,2 +1,11 @@
 # Ref: https://help.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository#configuring-the-template-chooser
 blank_issues_enabled: true  # default
+contact_links:
+- name: 💡 Feature Request
+  url: https://github.com/chanzuckerberg/napari-hub/discussions/new?category=ideas
+  about: |
+  Have an feature request? Start an "Ideas" thread on our discussion forum.
```
Comment (liaprins-czi) [inline .github/ISSUE_TEMPLATE/config.yml:7]: Tiny thing: "an" should be "a"

Comment (liaprins-czi) [reply .github/ISSUE_TEMPLATE/config.yml:7]: Also, it says 'start an "Ideas" thread on our discussion forum'... but should we start that general "Ideas" thread, and then link to that thread within the forum so it is easier for people to put things in the right place?

**Review by liaprins-czi — APPROVED** [review]
I added a comment, but overall looks good to me!

**Review by ziyangczi — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/config.yml (hunk 5 above)
Comment (neuromusic) [reply .github/ISSUE_TEMPLATE/config.yml:7]: >"Ideas" and "Q&A" threads are already types of threads that you can tag/categorizie as such

exactly. maybe I should reword to "Start a new thread on our discussion forum in the *Ideas* category."

is that more clear?

**Pull request merged**